gcc/testsuite/rust/compile/let-pattern.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    let ((_x, _y, _, _z)) = (1, 2, 3, 4);
```

Comment:
Can you make another test case:

```rust
fn main() -> i32 {
  let a,b = (1, 2);
  let c = a + b;
  c - a - b
}
```

I want to see the -fdump-tree-gimple. This is really important to get right so we need to be careful about merging this until we are happy.